### PR TITLE
MODCLUSTER-807 Change the mod_cluster license to ASL 2.0

### PR DESCRIPTION
**Pull request by rhusar**
Resolves
https://issues.redhat.com/browse/MODCLUSTER-807

### CODE REVIEW

File: container/spi/src/main/java/org/jboss/modcluster/container/Connector.java [hunk 1]
```diff
@@ -28,6 +11,7 @@
  *
  * @author Paul Ferraro
  */
+
```
Comment (jajik) [inline container/spi/src/main/java/org/jboss/modcluster/container/Connector.java:14]: Is this intentional?

File: container/tomcat-8.5/src/main/java/org/jboss/modcluster/container/tomcat/mbeans-descriptors.xml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 <?xml version="1.0"?>
+<!--
+  ~ Copyright The mod_cluster Project Authors
+  ~ SPDX-License-Identifier: Apache-2.0
+  -->
+
```
Comment (jajik) [inline container/tomcat-8.5/src/main/java/org/jboss/modcluster/container/tomcat/mbeans-descriptors.xml:6]: Here again (I mean the blank space – there's no blank line on most places).

File: container/tomcat-9.0/src/main/java/org/jboss/modcluster/container/tomcat/mbeans-descriptors.xml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 <?xml version="1.0"?>
+<!--
+  ~ Copyright The mod_cluster Project Authors
+  ~ SPDX-License-Identifier: Apache-2.0
+  -->
+
```
Comment (jajik) [inline container/tomcat-9.0/src/main/java/org/jboss/modcluster/container/tomcat/mbeans-descriptors.xml:6]: Same.

File: core/src/main/java/org/jboss/modcluster/config/ProxyConfiguration.java [hunk 4]
```diff
@@ -1,22 +1,6 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ * Copyright The mod_cluster Project Authors
+ * SPDX-License-Identifier: Apache-2.0
  */
```
Comment (jajik) [inline core/src/main/java/org/jboss/modcluster/config/ProxyConfiguration.java:5]: Same.

File: core/src/main/java/org/jboss/modcluster/config/builder/AbstractConfigurationBuilder.java [hunk 5]
```diff
@@ -1,23 +1,6 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ * Copyright The mod_cluster Project Authors
+ * SPDX-License-Identifier: Apache-2.0
  */
```
Comment (jajik) [inline core/src/main/java/org/jboss/modcluster/config/builder/AbstractConfigurationBuilder.java:5]: Same.

File: core/src/main/java/org/jboss/modcluster/config/builder/AdvertiseConfigurationBuilder.java [hunk 6]
```diff
@@ -1,23 +1,6 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ * Copyright The mod_cluster Project Authors
+ * SPDX-License-Identifier: Apache-2.0
  */
```
Comment (jajik) [inline core/src/main/java/org/jboss/modcluster/config/builder/AdvertiseConfigurationBuilder.java:5]: Same

File: core/src/main/java/org/jboss/modcluster/config/builder/BalancerConfigurationBuilder.java [hunk 7]
```diff
@@ -1,23 +1,6 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ * Copyright The mod_cluster Project Authors
+ * SPDX-License-Identifier: Apache-2.0
  */
```
Comment (jajik) [inline core/src/main/java/org/jboss/modcluster/config/builder/BalancerConfigurationBuilder.java:5]: Same

File: core/src/main/java/org/jboss/modcluster/config/builder/Creator.java [hunk 8]
```diff
@@ -1,23 +1,6 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ * Copyright The mod_cluster Project Authors
+ * SPDX-License-Identifier: Apache-2.0
  */
```
Comment (jajik) [inline core/src/main/java/org/jboss/modcluster/config/builder/Creator.java:5]: Same.

**Review by jajik — CHANGES_REQUESTED** [review]
Other than the http instead of httpS it looks correct.

I left there some comments about inconsistent blank spaces after comments, but that's not an issue.

File: pom.xml [hunk 9]
```diff
@@ -39,8 +22,8 @@
 
     <licenses>
         <license>
-            <name>GNU Lesser General Public License v3.0 or later</name>
-            <url>https://repository.jboss.org/licenses/lgpl-3.0.txt</url>
+            <name>Apache License 2.0</name>
+            <url>http://repository.jboss.org/licenses/apache-2.0.txt</url>
```
[the remark this answers is not in the document]
Comment (rhusar) [reply pom.xml:26]: I used the code from wildfly, but I see no reason to use http here either... i will fix this.

[rhusar@ribera tmp]$ curl --head http://repository.jboss.org/licenses/apache-2.0.txt
HTTP/1.1 301 Moved Permanently
Server: AkamaiGHost
Content-Length: 0
Location: https://repository.jboss.org/licenses/apache-2.0.txt
Expires: Thu, 07 Dec 2023 11:30:47 GMT
Cache-Control: max-age=0, no-cache, no-store
Pragma: no-cache
Date: Thu, 07 Dec 2023 11:30:47 GMT
Connection: keep-alive
x-rh-edge-request-id: 262299b7
x-rh-edge-reference-id: 0.7ef06e68.1701948647.262299b7
x-rh-edge-cache-status: NotCacheable from child

File: container/spi/src/main/java/org/jboss/modcluster/container/Connector.java (hunk 1 above)
Comment (rhusar) [reply container/spi/src/main/java/org/jboss/modcluster/container/Connector.java]: Not sure how that happened, fixed.

File: core/src/main/java/org/jboss/modcluster/config/builder/AdvertiseConfigurationBuilder.java (hunk 6 above)
Comment (rhusar) [reply core/src/main/java/org/jboss/modcluster/config/builder/AdvertiseConfigurationBuilder.java]: Let's go with no space after licence - fixed.

File: core/src/main/java/org/jboss/modcluster/config/builder/BalancerConfigurationBuilder.java (hunk 7 above)
Comment (rhusar) [reply core/src/main/java/org/jboss/modcluster/config/builder/BalancerConfigurationBuilder.java]: Let's go with no space after licence - fixed.

File: core/src/main/java/org/jboss/modcluster/config/builder/ConfigurationBuilder.java [hunk 10]
```diff
@@ -1,23 +1,6 @@
...
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this software; if not, write to the Free
- * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
- * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ * Copyright The mod_cluster Project Authors
+ * SPDX-License-Identifier: Apache-2.0
  */
```
[the remark this answers is not in the document]
Comment (rhusar) [reply core/src/main/java/org/jboss/modcluster/config/builder/ConfigurationBuilder.java]: Let's go with no space after licence - fixed.

File: pom.xml (hunk 9 above)
[the remark this answers is not in the document]
Comment (rhusar) [reply pom.xml]: Fixed.

**Review by jajik — CHANGES_REQUESTED** [review]

File: README.md [hunk 11]
```diff
@@ -109,9 +109,6 @@ Project mod_cluster uses Red Hat Jira issue tracker under MODCLUSTER project:
 
 [https://issues.redhat.com/browse/MODCLUSTER](https://issues.redhat.com/browse/MODCLUSTER)
 
-
 License
 -------
-
-This software is distributed under the terms of the GNU Lesser General Public License (see [LICENSE.txt](LICENSE.txt)).
-
+* [Apache License 2.0](http://www.apache.org/licenses/LICENSE-2.0)
```
[the remark this answers is not in the document]
Comment (rhusar) [reply README.md]: The licence itself references this URL, i.e.

Apache License
Version 2.0, January 2004
http://www.apache.org/licenses/

but I dont think that matters. Lets not be silly and use https  here too.

**Review by jajik — APPROVED** [review]
LGTM

**Comment by rhusar** [comment]
Thanks for the review @jajik

**Pull request merged**